alibi/explainers/similarity/metrics.py

```diff
@@ -0,0 +1,80 @@
+from typing import Union
+
+import numpy as np
+
+
+def dot(X: np.ndarray, Y: np.ndarray) -> Union[float, np.ndarray]:
+    """
+    Performs a dot product between the vector(s) in X and vector Y.
+
+    Parameters
+    ----------
+    X:
+        Matrix of vectors.
+    Y:
+        Single vector
+
+    Returns
+    -------
+        Dot product between the vector(s) in X and vector Y.
+    """
+    if len(X.shape) == 1:
+        assert X.shape == Y.shape, "The vector `X` and `Y` need to have the same dimensions."
```

Comment:
I'm assuming these are asserts rather than errors because they should never happen unles something is logically wrong with our implementation?

Reply:
Yeah, these are here for our sake. Can remove if preferred though?

Reply:
Fine by me to leave.